what4/src/What4/Protocol/SMTLib2.hs

```diff
@@ -662,7 +665,7 @@ writeCheckSat :: SMTLib2Tweaks a => WriterConn t (Writer a) -> IO ()
 writeCheckSat w = addCommandNoAck w SMT2.checkSat
 
 writeExit :: forall a t. SMTLib2Tweaks a => WriterConn t (Writer a) -> IO ()
-writeExit w = addCommand w SMT2.exit
+writeExit w = mapM_ (addCommand w) (smtlib2exitCommand @a)
```

Comment:
I don't understand what's happening with this change.

Reply:
ABC crashes if you give it an `(exit)` command, so we need the option for this list to be empty. It could alternatively be a `Maybe` instead of a list.

Reply:
Oh, I see. It is already a `Maybe`, but my confusion was not realising that the definition above is a default definition in a class.